### PR TITLE
Feature/animation-on-click

### PR DESCRIPTION
**Pull request by WellsSA**
# Animation on Click

This PR changes the static infinite shooting animation for the gun and bullets so we can trigger them on demand, allowing us to: 
- trigger the gun animation on user click
- "shoot" new bullets for every click

## Checklist features
- added public/animation.js file for controlling animations via JS
- changed main HTML structure for both gun and bullets
- Added `click` handler to `html` for triggering shotgun animation on user click
- Added bullet cloning for multiple shots
- Allowed animation restart on re-click
- Added gun trigger animation
- Changed gun barrel animation 
- Vertically rotated orientation for better usage on mobile

## Preview

https://user-images.githubusercontent.com/41883467/203111169-f5d456aa-8db0-4284-b4d1-180cf6b1230d.mov

### CODE REVIEW

File: mobile-controller/public/styles.css [hunk 1]
```diff
@@ -59,7 +63,7 @@ html{
   position:absolute;
   top:170px;
   left:120px;
-  animation: Peww 1.1s infinite;
+  /* animation: Peww 1.1s infinite; */
```
Comment (ErickWendel) [inline mobile-controller/public/styles.css]: would you remove all those unnecessary codes?

File: mobile-controller/public/src/index.js [hunk 2]
```diff
@@ -108,8 +109,8 @@ class App {
   }
 
   setupShooter() {
-
-    window.document.body.onclick = () => {
+    window.document.querySelector('html').onclick = () => {
```
Comment (ErickWendel) [inline mobile-controller/public/src/index.js:112]: why did you change from body to HTML?

**Review by ErickWendel — CHANGES_REQUESTED** [review]

File: mobile-controller/public/src/index.js (hunk 2 above)
Comment (WellsSA) [reply mobile-controller/public/src/index.js:112]: I changed it because as we rotate the body orientation to make the gun vertical, the click handler doesn't work the same. So to keep the click working in the entire page I got the body's parent element which is HTML. 

We can change the click handler back to body if we rotate a different element, like a `#root` container or so.

Comment (ErickWendel) [reply mobile-controller/public/src/index.js:112]: got it!

File: mobile-controller/public/styles.css (hunk 1 above)
Comment (WellsSA) [reply mobile-controller/public/styles.css]: Yessir, I left them in purpose to keep tracking of which animation is referenced where, because the naming convention `item#2 { animation: Peww }`, `item#4 { animation: Pewww2 }` can be confusing. 

However, I just noticed now that a better approach is to keep these animations referenced on CSS via a custom class, and just toggle the class on JS, instead of toggle the animations.

Comment (ErickWendel) [reply mobile-controller/public/styles.css]: nice! Will you do this change?

File: mobile-controller/public/src/index.js (hunk 2 above)
Comment (WellsSA) [reply mobile-controller/public/src/index.js:112]: I changed it to allow shooting anywhere in the screen. With the handler on `body` it was only firing the gun when you click anywhere in the gun. Like this:

https://user-images.githubusercontent.com/41883467/203633585-798172bd-67c8-4075-b836-73dbdaa1d684.mov

This happens because when you have the click handler on body, the click is only perceived when the `propagation` is from an element inside the body. (which means if you click an element that is in the body, the click propagates and reaches the body, and it triggers your event handler.
However, your background is outside the body (on the parent element, which is html)
https://github.com/WellsSA/DuckHunt-JS/blob/d3d904916f1311d9470dc034be5a6ef9b33b7a73/mobile-controller/public/styles.css#L1-L2

So I changed the handler to html and the result is this:

https://user-images.githubusercontent.com/41883467/203633936-704f99e4-3d1a-44d8-9ed1-eecad2f73de3.mov

File: mobile-controller/public/styles.css (hunk 1 above)
Comment (WellsSA) [reply mobile-controller/public/styles.css]: Yessir! Done

**Review by ErickWendel — APPROVED** [review]

**Comment by ErickWendel** [comment]
Landed! Thanks for your contribution my friend!
